### PR TITLE
[Outlook] (sensitivity label) Create sample

### PR DESCRIPTION
**Pull request by samantharamon**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                              |
| New feature?    | yes                           |
| New sample?     | yes                             |
| Related issues? | N/A |

## What's in this Pull Request?

This sample uses the sensitivity label API in an event-based add-in to verify and apply the **Highly Confidential** sensitivity label to outgoing messages that contain at least one attachment or a recipient who's a member of the fictitious Fabrikam legal team. When the **Highly Confidential** label is applied, a fictitious legal hold account is added to the **Bcc** field of the message.

## Guidance

See above.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ All of the following samples show how to access and work with a user's Microsoft
 | [Use Outlook event-based activation to indicate external recipients](Samples/outlook-tag-external) | Uses event-based activation to run an Outlook add-in when the user changes recipients while composing a message. The add-in also uses the appendOnSendAsync API to add a disclaimer.|
 | [Use Outlook event-based activation to set the signature](Samples/outlook-set-signature) | Uses event-based activation to run an Outlook add-in when the user creates a new message or appointment.|
 | [Use Outlook Smart Alerts](Samples/outlook-check-item-categories/) | Uses Outlook Smart Alerts to verify that required color categories are applied to a new message or appointment before it's sent.|
+| [Verify the sensitivity label of a message (preview)](Samples/outlook-check-item-categories/) | Uses the sensitivity label API in an event-based add-in to verify and apply the **Highly Confidential** sensitivity label to applicable outgoing messages. |
```
Comment (ElizabethSamuel-MSFT) [inline README.md:57]:
```suggestion
| [Verify the sensitivity label of a message (preview)](Samples/outlook-verify-sensitivity-label/) | Uses the sensitivity label API in an event-based add-in to verify and apply the **Highly Confidential** sensitivity label to applicable outgoing messages. |
```

File: Samples/outlook-verify-sensitivity-label/README.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+Office.context.mailbox.item.sensitivityLabel.setAsync(highlyConfidentialLabel, { asyncContext: event }, (result) => {
+    const event = result.asyncContext;
+    if (result.status === Office.AsyncResultStatus.Failed) {
+        console.log("Unable to set the Highly Confidential sensitivity label to the message.");
+        console.log(`Error: ${result.error.message}`);
+        event.completed({ allowEvent: false, errorMessage: "Unable to set the Highly Confidential sensitivity label to the message. Save your message, then restart Outlook." });
+        return;
+    }
+    
+    event.completed({ allowEvent: false, errorMessage: "Due to the contents of your message, the sensitivity label has been set to Highly Confidential and the Legal Hold account has been added to the Bcc field.\nTo learn more, see Fabrikam's information protection policy.\n\nDo you need to make changes to your message?" });
```
Comment (ElizabethSamuel-MSFT) [inline Samples/outlook-verify-sensitivity-label/README.md:285]: Do we know if `\n` renders correctly on Mac?

File: Samples/outlook-verify-sensitivity-label/README.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+|Version|Date|Comments|
+|-------|----|--------|
+|1.0|March 27, 2023|Initial release|
+
+## Copyright
+
+Copyright (c) 2023 Microsoft Corporation. All rights reserved.
+
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information, see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
```
Comment (davidchesnut) [inline Samples/outlook-verify-sensitivity-label/README.md:311]:
```suggestion
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information, see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

<img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/outlook-verify-sensitivity-label" />
```

**Review by davidchesnut — CHANGES_REQUESTED** [review]
Looks great! Please take a look at a few comments I had. Thanks!

File: Samples/outlook-verify-sensitivity-label/README.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+description: "Learn how to verify and update the sensitivity label of a message using an event-based add-in."
+---
+
+# Verify the sensitivity label of a message (preview)
+
+**Applies to**: Outlook on Windows | Outlook on Mac | modern Outlook on the web
+
+## Summary
+
+This sample uses the sensitivity label API in an event-based add-in to verify and apply the **Highly Confidential** sensitivity label to outgoing messages that contain at least one attachment or a recipient who's a member of the fictitious Fabrikam legal team. When the **Highly Confidential** label is applied, a fictitious legal hold account is added to the **Bcc** field of the message.
```
[the remark this answers is not in the document]
Comment (samantharamon) [reply Samples/outlook-verify-sensitivity-label/README.md:25]: Thanks for catching this. Yes, this should be lowercase.

File: Samples/outlook-verify-sensitivity-label/README.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+1. Create a new message.
+1. Do one of the following:
+
+    - Add an attachment to the message.
+    - Add the email address of a fictitious Fabrikam legal team member to the **To**, **Cc**, or **Bcc** field using the format, **-legal@fabrikam.com**. For example, **eli-legal@fabrikam.com**.
+
+1. (Optional) Add a subject or content to the body of the message.
+1. Select **Send**.
+
+    The sensitivity label of the message is set to **Highly Confidential** and the **legalhold@fabrikam.com** account is added to the **Bcc** field. A Smart Alerts dialog appears that reads, "Due to the contents of your message, the sensitivity label has been set to Highly Confidential and the Legal Hold account has been added to the **Bcc** field. To learn more, see Fabrikam's information protection policy. Do you need to make changes to your message?"
```
[the remark this answers is not in the document]
Comment (samantharamon) [reply Samples/outlook-verify-sensitivity-label/README.md]: I tested this with just an attachment and confirmed that the legalhold@fabrikam.com account was added to the Bcc field. If you get a chance, could you try testing that scenario again? Let me know if you're still able to replicate the issue. Thanks!

[the remark this answers is not in the document]
Comment (samantharamon) [reply Samples/outlook-verify-sensitivity-label/README.md]: Thanks for troubleshooting this and finding the fix, @davidchesnut! The `associate` calls have now been moved to `Office.onReady`.

**Review by davidchesnut — APPROVED** [review]
Looks great! Thanks!

File: Samples/outlook-verify-sensitivity-label/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (davidchesnut) [reply Samples/outlook-verify-sensitivity-label/README.md]: Fixed now, thanks!

File: Samples/outlook-verify-sensitivity-label/README.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+- office-add-ins
+- office-outlook
+- office
+languages:
+- javascript
+extensions:
+  contentType: samples
+  technologies:
+  - Add-ins
+  createdDate: 03/27/2023 10:00:00 AM
```
Comment (ElizabethSamuel-MSFT) [inline Samples/outlook-verify-sensitivity-label/README.md:15]:
```suggestion
  createdDate: 04/18/2023 10:00:00 AM
```

File: Samples/outlook-verify-sensitivity-label/README.md [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+
+|Solution|Authors|
+|--------|-------|
+|Verify the sensitivity label of a message using an event-based add-in.|Microsoft|
+
+## Version history
+
+|Version|Date|Comments|
+|-------|----|--------|
+|1.0|March 27, 2023|Initial release|
```
Comment (ElizabethSamuel-MSFT) [inline Samples/outlook-verify-sensitivity-label/README.md:305]:
```suggestion
|1.0|April 18, 2023|Initial release|
```

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Pull request merged**